### PR TITLE
Update to xclim 0.31

### LINKED ISSUE #193: Upgrade to xclim 0.29

### PR DESCRIPTION
**Pull request by aulemahal**
## Overview

This PR fixes #193.

Changes:

* Some modifications on how the default value of xclim indicator's parameters is handled.
* Addition of the "boolean" type to the list of supported types (for the new `bootstrap` arg)
* Addition of `tas_25, tas_75, pr_25 and pr_75` as supported variables for the cold/warm dry/wet indicators.
* Modification on how sdba objects are created.
* Pin xESMF < 0.6.1 to circumvent a bug in `average_shape` of clisops. This version is also broken in some other cases, so it's better to avoid.  Pin ESMpy to <8.2.0 as there is a bug with xESMF in that version.
* Adapt the ensemble process creation to use the new `Indicator.parameters`.

## Related Issue / Discussion
Superseeds #200.

## Additional Information

I also harmonized the requirements by removing any "dev" deps from `environment.yml` (that's the role of `requirements_dev.txt`) and made it so that `environment.yml` has the same packages than `requirements.txt`.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -12,15 +12,18 @@ dependencies:
...
-  - xclim ==0.27
+  - xclim == 0.31
   - clisops >=0.6.4
   - pywps ==4.4.5
   - parse
-  - pandoc  # for building docs on Travis-CI, version on Pypi too old
-  - xesmf >=0.5.3
-  - nbconvert >6.0  # see: https://github.com/jupyter/jupyter_client/issues/637
-  - jupyter_client >7.0  # see: https://github.com/jupyter/jupyter_client/issues/637
+  - xesmf >=0.5.3,!=0.6.1  # lclisops 0.7 bugs with xESMF 0.6.1
```
Comment (huard) [inline environment.yml:28]:
```suggestion
  - xesmf >=0.5.3,!=0.6.1  # clisops 0.7 bugs with xESMF 0.6.1
```

**Review by huard — APPROVED** [review]

File: docs/source/notebooks/finch-usage.ipynb [hunk 2]
```diff
@@ -59,7 +59,7 @@
       "Help on method frost_days in module birdy.client.base:\n",
       "\n",
       "frost_days(tasmin=None, thresh='0 degC', freq='YS', check_missing='any', missing_options=None, cf_compliance='warn', data_validation='raise', variable=None) method of birdy.client.base.WPSClient instance\n",
-      "    Number of days where daily minimum temperatures are below 0.\n",
+      "    Number of days where daily minimum temperatures are below a threshold temperature.\n",
```
Comment (tlvu) [inline docs/source/notebooks/finch-usage.ipynb:62]: This change breaks Jenkins see http://jenkins.ouranos.ca/job/PAVICS-e2e-workflow-tests/job/master/1309/console.

Do we need a new Finch to be deployed to the server?

Else add `# NBVAL_IGNORE_OUTPUT` and we accept the Finch deployed in prod do not match our latest notebooks that is autodeployed to prod.

File: docs/source/notebooks/finch-usage.ipynb [hunk 3]
```diff
@@ -558,10 +558,10 @@
...
        "    standard_name:  days_with_air_temperature_below_threshold\n",
-       "    long_name:      Number of frost days (tmin &lt; 0c)\n",
-       "    description:    Annual number of days with minimum daily temperature belo...</pre><div class='xr-wrap' hidden><div class='xr-header'><div class='xr-obj-type'>xarray.DataArray</div><div class='xr-array-name'>'frost_days'</div><ul class='xr-dim-list'><li><span class='xr-has-index'>time</span>: 20</li><li><span class='xr-has-index'>lat</span>: 6</li><li><span class='xr-has-index'>lon</span>: 7</li></ul></div><ul class='xr-sections'><li class='xr-section-item'><div class='xr-array-wrap'><input id='section-3fdbff45-0f19-4cf0-9bd4-9c79128fee37' class='xr-array-in' type='checkbox' checked><label for='section-3fdbff45-0f19-4cf0-9bd4-9c79128fee37' title='Show/hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-array-preview xr-preview'><span>122.0 121.0 98.0 47.0 33.0 19.0 ... 249.0 229.0 193.0 156.0 103.0</span></div><div class='xr-array-data'><pre>array([[[122., 121., ...,  19.,   9.],\n",
+       "    long_name:      Number of frost days (tmin &lt; 0 degc)\n",
```
Comment (tlvu) [inline docs/source/notebooks/finch-usage.ipynb:563]: This change also breaks Jenkinks, http://jenkins.ouranos.ca/job/PAVICS-e2e-workflow-tests/job/master/1309/console

Same for the `xclim_history:` to `history:`.

File: docs/source/notebooks/finch-usage.ipynb (hunk 2 above)
Comment (huard) [reply docs/source/notebooks/finch-usage.ipynb:62]: Yes, we should release Finch.

Comment (tlvu) [reply docs/source/notebooks/finch-usage.ipynb:62]: > Yes, we should release Finch.

Can we do this soon?  Jenkins broken daily in prod.

Comment (huard) [reply docs/source/notebooks/finch-usage.ipynb:62]: On it.

Comment (tlvu) [reply docs/source/notebooks/finch-usage.ipynb:62]: https://github.com/bird-house/birdhouse-deploy/pull/223

**Pull request merged**